packages/esm-patient-registration-app/src/patient-registration/field/id/id-field.test.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { render, screen, fireEvent } from '@testing-library/react';
+import { Identifiers } from './id-field.component';
+import { Resources, ResourcesContext } from '../../../offline.resources';
+import { Form, Formik } from 'formik';
+import { PatientRegistrationContext } from '../../patient-registration-context';
+import { useConfig } from '@openmrs/esm-framework';
+import { openmrsID } from '../__mocks__/identifiers.mock';
+import { mockedIdentifierTypes } from '../__mocks__/identifier-types.mock';
+
+jest.mock('@openmrs/esm-framework', () => ({
+  ...jest.requireActual('@openmrs/esm-framework'),
+  useConfig: jest.fn(),
+}));
+
+describe('Identifiers', () => {
+  const mockResourcesContextValue = {
+    addressTemplate: [],
+    currentSession: {
+      authenticated: true,
+      sessionId: 'JSESSION',
+      currentProvider: { uuid: 'provider-uuid', identifier: 'PRO-123' },
+    },
+    relationshipTypes: [],
+    identifierTypes: [...mockedIdentifierTypes],
+  } as Resources;
+
+  (useConfig as jest.Mock).mockImplementation(() => ({
+    defaultPatientIdentifierTypes: ['OpenMRS ID'],
+  }));
+
+  it('should render loading skeleton when identifier types are loading', () => {
+    render(
+      <ResourcesContext.Provider value={[]}>
+        <Formik initialValues={{}} onSubmit={null}>
+          <Form>
+            <PatientRegistrationContext.Provider
+              value={{
+                setFieldValue: jest.fn(),
+                initialFormValues: { identifiers: { openmrsID } },
+                setInitialFormValues: jest.fn(),
+                values: {
+                  identifiers: { openmrsID },
+                },
+              }}>
+              <Identifiers />
+            </PatientRegistrationContext.Provider>
+          </Form>
+        </Formik>
+      </ResourcesContext.Provider>,
+    );
+    expect(screen.getByTestId('loading-skeleton')).toBeInTheDocument();
+  });
+
+  it('should render identifier inputs when identifier types are loaded', () => {
+    render(
+      <ResourcesContext.Provider value={mockResourcesContextValue}>
+        <Formik initialValues={{}} onSubmit={null}>
+          <Form>
+            <PatientRegistrationContext.Provider
+              value={{
+                setFieldValue: jest.fn(),
+                initialFormValues: { identifiers: { openmrsID } },
+                setInitialFormValues: jest.fn(),
+                values: {
+                  identifiers: { openmrsID },
```

Comment:
when I am changing this openmrsID. then the tests start to fail.

> TypeError: Cannot read properties of null (reading 'identifierTypeUuid')

      24 |   const { isOffline, values, setFieldValue } = useContext(PatientRegistrationContext);
      25 |   const identifierType = useMemo(
    > 26 |     () => identifierTypes.find((identifierType) => identifierType.uuid === patientIdentifier.identifierTypeUuid),
         |                                                                                              ^
      27 |     [patientIdentifier, identifierTypes],
      28 |   );
      29 |   const { autoGeneration, initialValue, identifierValue, identifierName, required } = patientIdentifier;